### DISCUSSION TITLE
What's an idiom for what to do when routing to content containing a component?

### DISCUSSION THREAD

**Discussion by quimt** [HapticX/happyx#363] [Q&A]
Sometimes with external apps I need to ensure that they render correctly after routing to them. How can I ask a component to do something whenever changing to its route(s)?

Perhaps an `@routed` lifecycle would be helpful to give direct control over how components handle route transitions.

**Comment by Ethosa** [comment]
Tracking route changes is not currently implemented. However, you can use the `currentRoute` variable to find out the current route
